src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,288 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "PersistentStorageOperationalKeystoreHSM.h"
+#include "CHIPCryptoPALHsm_SE05X_utils.h"
+#include <crypto/hsm/CHIPCryptoPALHsm.h>
+
+#if ENABLE_HSM_GENERATE_EC_KEY
+
+namespace chip {
+
+using namespace chip::Crypto;
+
+#define MAX_KEYID_SLOTS_FOR_FABRICS 32
+#define FABRIC_SE05X_KEYID_START 0x7D100000
+
+struct keyidFabIdMapping_t
+{
+    uint32_t keyId;
+    FabricIndex fabricIndex;
+    bool isPending;
+    Crypto::P256KeypairHSM * pkeyPair;
+} keyidFabIdMapping[MAX_KEYID_SLOTS_FOR_FABRICS] = {
+    0,
+};
+
+uint8_t getEmpytSlotId()
+{
+    uint8_t i = 0;
+    while (i < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].keyId == 0 && keyidFabIdMapping[i].isPending == 0)
+        {
+            break;
+        }
+        i++;
+    }
+    return i;
+}
+
+void ResetPendingSlot(uint32_t slotId)
+{
+    if (slotId < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        keyidFabIdMapping[slotId].pkeyPair    = nullptr;
+        keyidFabIdMapping[slotId].keyId       = 0;
+        keyidFabIdMapping[slotId].fabricIndex = 0;
+        keyidFabIdMapping[slotId].isPending   = 0;
+    }
+}
+
+bool PersistentStorageOperationalKeystoreHSM::HasOpKeypairForFabric(FabricIndex fabricIndex) const
+{
+    uint8_t i = 0;
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), false);
+
+    // If there was a pending keypair, then there's really a usable key
+    if (mIsPendingKeypairActive && (fabricIndex == mPendingFabricIndex) && (mPendingKeypair != nullptr))
+    {
+        ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> mPendingKeypair found");
+        return true;
+    }
+
+    while (i < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].fabricIndex == fabricIndex)
+        {
+            ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> stored keyPair found");
+            return true;
+        }
+        i++;
+    }
+
+    ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> No key found");
+    return false;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::NewOpKeypairForFabric(FabricIndex fabricIndex,
+                                                                          MutableByteSpan & outCertificateSigningRequest)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint8_t slotId = getEmpytSlotId();
+
+    VerifyOrReturnError(slotId < MAX_KEYID_SLOTS_FOR_FABRICS, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    ChipLogProgress(Crypto, "SE05x: New Op Keypair for Fabric %02x", fabricIndex);
+
+    // Replace previous pending keypair, if any was previously allocated
+    ResetPendingKey();
+    if (mPendingKeypair != nullptr)
```

Comment:
This doesn't make any sense.  `mPendingKeypair` is _always_ null here.

Reply:
Yes. Fixed this. Moved ResetPendingSlot to ResetPendingKey